### PR TITLE
[Tooling] Automate distribution of TestFlight builds to external tester groups

### PR DESCRIPTION
**Pull request by AliSoftware**
What it says on the tin.

As part of calling `upload_to_test_flight` at the end of release builds (post code-freeze, beta, or final release), the automation will now automatically add "external tester groups" to the build in TestFlight for you.

It will use the current content of the entries from the `CHANGELOG.md`[^1] for the "What to test" field in TestFlight when submitting to external testing, after removing the GitHub links from the bullet points.

Once this lands, I'll open a diff to remove the corresponding manual tasks from the MC Release Scenario, as they would not be necessary to carry out manually anymore:

 - "[ ] Watch for the `#pocket-casts-dev-ios` notification stating that the build has completed, then submit/distribute the build to testers on TestFlight](https://appstoreconnect.apple.com/apps/414834813/testflight/ios)" _(Code Freeze and Intermediate Betas)_
 - "[ ] Go to [App Store Connect](https://appstoreconnect.apple.com/apps/414834813/testflight/ios). Submit for testing review. Then, test with this build installed over current App Store build." _(Final Build / `finalize_release`)_

This PR also ensure that the GitHub Releases now use the latest content of the `CHANGELOG.md`[^1] (this time keeping the GitHub links).

[^1]: The automation now re-extracting the content of the `CHANGELOG.md` at the time the betas are created—instead of using on the ones that were extracted into `release_notes.txt` during code freeze—ensures that those will include any additional entries that might have been added to the `CHANGELOG.md` post code-freeze, e.g. entries for betafixes that landed in `release/*` since.

## To test

Not sure we can easily test this without doing an actual new beta.

I'm currently making this PR target `trunk` because I'll be on Support Rotation next week and AFK until EOY after that and thus unavailable to provide support in case anything would go wrong when testing those changes in production.

However, if there is a new beta planned for PCiOS before end of this week, I'd love to have this change cherry-picked onto `release/7.79` so that we can use the occasion to validate it works as part of the beta (and I can address any issue if needed or revert if not enough time to address them) before adopting those improvements for the next MC Release Scenario sooner than later 🙂

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -25,6 +25,15 @@ USER_ENV_FILE_PATH = File.join(Dir.home, '.a8c-apps', ENV_FILE_NAME)
 
 TEAM_ID_APP_STORE = 'PZYM8XX95Q'
 TEAM_ID_ENTERPRISE = '99KV9Z6BKV'
+EXTERNAL_BETA_TESTER_GROUPS = [
+  'A8cs',
+  'Alpha Slack',
+  'Beta Slack',
+  'Stable Slack',
+  'Reddit',
+  'Big Databases',
```
Comment (spencertransier) [inline fastlane/Fastfile]:
```suggestion
  'Big Databases',
  'Reddit',
  'Stable Slack',
```

Just alphabetizing 🙂

File: fastlane/Fastfile [hunk 2]
```diff
@@ -551,7 +560,7 @@ platform :ios do
     zip(path: APP_STORE_CONNECT_XCARCHIVE_PATH, output_path: APP_STORE_CONNECT_XCARCHIVE_ZIP_PATH)
   end
 
-  # Distributes the binary to TestFlight
+  # Uploads the binary to TestFlight and distribute it to testers
```
Comment (spencertransier) [inline fastlane/Fastfile]:
```suggestion
  # Uploads the binary to TestFlight and distributes it to testers
```

File: Gemfile [hunk 3]
```diff
@@ -9,7 +9,7 @@ gem 'danger-dangermattic', '~> 1.0'
 gem 'fastlane', '~> 2.216'
 gem 'fastlane-plugin-appcenter', '~> 2.1'
 gem 'fastlane-plugin-sentry', '~> 1.14'
-gem 'fastlane-plugin-wpmreleasetoolkit', '~> 12.2'
+gem 'fastlane-plugin-wpmreleasetoolkit', '~> 12.4'
```
Comment (AliSoftware) [inline Gemfile:12]: ℹ️  This update is in particular to be able to benefit from [this fix on `extract_release_notes_for_version`](https://github.com/wordpress-mobile/release-toolkit/pull/623)

File: fastlane/Fastfile (hunk 1 above)
Comment (AliSoftware) [reply fastlane/Fastfile]: Good point… though that means that `* Slack` groups are not together anymore now 😄 

That being said, the groups are also listed alphabetically in App Store Connect, so maybe that still make sense to alphabetize them here too after all 🙂

**Review by spencertransier — APPROVED** [review]
Looks good! I verified the names of the external tester group names and that the regex is properly removing the GitHub links from the changelog.

**Comment by AliSoftware** [comment]
Thanks for the quick review and testing @spencertransier !

⏸️  FYI I'll hold on merging this PR until I have a diff for the MC Release Scenario ready to land with corresponding changes (i.e. removal of now-obsolete manual tasks)

**Comment by leandroalonso** [comment]
@AliSoftware quick question unrelated to this change: are we removing the Slack notifications? While we don't need to follow the steps anymore it would be cool to keep receiving them because it's a signal everything went smoothly. :)

**Comment by AliSoftware** [comment]
> are we removing the Slack notifications? While we don't need to follow the steps anymore it would be cool to keep receiving them because it's a signal everything went smoothly. :)

There's no plan to remove those notifications 🙂 
In fact, [I've recently updated the message of those notifications to remove the instructions from the message](https://github.com/Automattic/pocket-casts-ios/pull/2533/commits/11babf1cde78845e05b1029f985c1e7620a3ee45) (because those instructions were already in the Release Scenario anyway, and better only keep a single source of truth).

So they'll still be sent, they'll just be shorter since without the instructions anymore 🙂 

_Note that even once this PR lands, steps for submitting the final build for review to ASC are still manual for now, as for those you need to copy the final release notes from the P2 comment while submitting and CI don't have them. I plan to automate that submission too in a follow-up PR (prompting you to paste the copy from P2 when you'll run the lane)… though not sure I will have time to land that one before I go EOY-AFK this Friday… we'll see)_

**Comment by AliSoftware** [comment]
ℹ️  Corresponding Scenario diff created (internal ref: D168562-code). Once it's approved I'll land both this PR and the scenario diff in concert.

**Pull request merged**